### PR TITLE
Revert "Allow pipelines from AutoMLSearch to be pickled  (#1721)"

### LINKED ISSUE #1912: AutoMLSearch get_pipeline always returns pipelines with the same name
Repro:

```python
from evalml.automl import AutoMLSearch
from evalml.demos import load_breast_cancer

X, y = load_breast_cancer()
automl = AutoMLSearch(X, y, problem_type="binary", max_batches=1)
automl.search()

pipelines = [automl.get_pipeline(i) for i in range(3)]
assert [p.name for p in pipelines] == ['LightGBM Classifier w/ Imputer', 'LightGBM Classifier w/ Imputer', 'LightGBM Classifier w/ Imputer']
```

The pipelines should not all have the same name. The estimators are different:

```python
[p.estimator.name for p in pipelines]
```

```python
['Baseline Classifier', 'Decision Tree Classifier', 'LightGBM Classifier']
```

### PR DESCRIPTION
**Pull request by jeremyliweishih**
Fixes #1912.

### CODE REVIEW

File: docs/source/user_guide/automl.ipynb
Comment (jeremyliweishih) [inline docs/source/user_guide/automl.ipynb:50]: Weird replacement here done by git but it looks fine: https://feature-labs-inc-evalml--1958.com.readthedocs.build/en/1958/user_guide/automl.html#AutoML-in-EvalML

Comment (freddyaboulton) [reply docs/source/user_guide/automl.ipynb:50]: I've been seeing it come up in other PRs as well. I don't know what's up but agree it doesn't look like a problem.

File: docs/source/user_guide/automl.ipynb
Comment (freddyaboulton) [inline docs/source/user_guide/automl.ipynb:367]: `.save` should still work right? Maybe we should add back this code snippet, but I agree we should delete `pickle.dumps`.

File: docs/source/user_guide/automl.ipynb
Comment (angela97lin) [reply docs/source/user_guide/automl.ipynb:50]: Same, looks fine though.

File: evalml/tests/automl_tests/test_automl_search_classification.py
Comment (freddyaboulton) [inline evalml/tests/automl_tests/test_automl_search_classification.py:708]: @jeremyliweishih I agree we should delete these tests but can we add coverage that `get_pipeline` works as expected? 

I think it can be in `test_automl.py` and it can be simple and only test one problem type. I think just checking against the rankings table should suffice:

```python
for _, ranking in automl.rankings.iterrows():
    pl = automl.get_pipeline(ranking.id)
    assert pl.parameters == ranking.parameters
    assert pl.name == ranking.pipeline_name
```

File: evalml/tests/automl_tests/test_automl.py
Comment (freddyaboulton) [inline evalml/tests/automl_tests/test_automl.py:1386]: Can we keep `train_best_pipeline=False`?

**Review by angela97lin — APPROVED** [review]
:shipit: ⛩️ !

**Review by freddyaboulton — APPROVED** [review]
@jeremyliweishih Looks good to me! Thanks for doing this. I think adding some coverage is worth it but not blocking merge.

File: docs/source/user_guide/automl.ipynb
Comment (dsherry) [reply docs/source/user_guide/automl.ipynb:50]: That's so weird. This is [the unicode character "right single quotation mark"](https://www.fileformat.info/info/unicode/char/2019/index.htm) 😆  perhaps jupyter changed the way they represent unicode somehow.

File: docs/source/user_guide/automl.ipynb
Comment (dsherry) [reply docs/source/user_guide/automl.ipynb:367]: Agreed, `save` still works because it uses `cloudpickle`, not `pickle`

File: evalml/tests/automl_tests/test_automl.py
Comment (dsherry) [reply evalml/tests/automl_tests/test_automl.py:1386]: +1

**Review by dsherry — APPROVED** [review]
Looking good! I didn't have any comments but I agree with @freddyaboulton 's comments about the tests and the automl user guide. We should address the test comments at least before merge.

**Review by bchen1116 — APPROVED** [review]
🚀

**Pull request merged**
